gateway-shell/src/main/java/org/apache/knox/gateway/shell/KnoxSession.java

```diff
@@ -570,6 +584,121 @@ public String toString() {
     return String.format(Locale.ROOT, "KnoxSession{base='%s'}", base);
   }
 
+  /**
+   * Persist provided Map to a file within the {user.home}/.knoxshell directory
+   * @param <T>
+   * @param fileName of persisted file
+   * @param map to persist
+   */
+  public static <T> void persistToKnoxShell(String fileName, Map<String, List<T>> map) {
+    String s = JsonUtils.renderAsJsonString(map);
+    String home = System.getProperty("user.home");
+    synchronized(KnoxSession.class) {
+      try {
+        write(new File(
```

Comment:
nit: might make sense to make write synchronized. to avoid calling that method without synchronization. Or at least move the synchronized block into the write method.